rpc/core/routes.go

```diff
@@ -37,25 +40,58 @@ func (env *Environment) GetRoutes() RoutesMap {
 		"consensus_params":     rpc.NewRPCFunc(env.ConsensusParams, "height", true),
 		"unconfirmed_txs":      rpc.NewRPCFunc(env.UnconfirmedTxs, "limit", false),
 		"num_unconfirmed_txs":  rpc.NewRPCFunc(env.NumUnconfirmedTxs, "", false),
+	}
+}
+
+func (env *Environment) SubscribeRoutes() RoutesMap {
+	return RoutesMap{
+		// subscribe/unsubscribe are reserved for websocket events.
+		"subscribe":       rpc.NewWSRPCFunc(env.Subscribe, "query"),
+		"unsubscribe":     rpc.NewWSRPCFunc(env.Unsubscribe, "query"),
+		"unsubscribe_all": rpc.NewWSRPCFunc(env.UnsubscribeAll, ""),
+	}
+}
 
-		// tx broadcast API
+func (env *Environment) BroadcastTxRoutes() RoutesMap {
+	// tx broadcast API
+	return RoutesMap{
 		"broadcast_tx_commit": rpc.NewRPCFunc(env.BroadcastTxCommit, "tx", false),
 		"broadcast_tx_sync":   rpc.NewRPCFunc(env.BroadcastTxSync, "tx", false),
 		"broadcast_tx_async":  rpc.NewRPCFunc(env.BroadcastTxAsync, "tx", false),
+	}
+}
+
+func (env *Environment) EvidenceRoutes() RoutesMap {
+	// evidence API
+	return RoutesMap{
+		"broadcast_evidence": rpc.NewRPCFunc(env.BroadcastEvidence, "evidence", false),
+	}
+}
 
-		// abci API
+func (env *Environment) ABCIQueryRoutes() RoutesMap {
+	// abci API
+	return RoutesMap{
 		"abci_query": rpc.NewRPCFunc(env.ABCIQuery, "path,data,height,prove", false),
 		"abci_info":  rpc.NewRPCFunc(env.ABCIInfo, "", true),
-
-		// evidence API
-		"broadcast_evidence": rpc.NewRPCFunc(env.BroadcastEvidence, "evidence", false),
 	}
 }
 
 // AddUnsafeRoutes adds unsafe routes.
-func (env *Environment) AddUnsafe(routes RoutesMap) {
+func (env *Environment) UnsafeRoutes() RoutesMap {
```

Comment:
API breaking, will remove.

Reply:
lets allow breaking changes here. If a user is using this API then god speed.